### DISCUSSION TITLE
looking forward to sprites

### DISCUSSION THREAD

**Discussion by michaelsarr** [picocomputer/community#41] [Q&A]
been playing around with some of the 320x240 graphics and working on my c chops
I really have not done a lot of c in the last few years

I'm really looking forward to the sprites, these were pretty cool on the atari (player/missle) and the c64 back in the day

here I have some functions to draw primitive shapes in the vga, just  pass in coordinates, radius & color
or start and end coordinates

I think my atari had some functions like plot and drawto so this took me back to the late 70s

this thing really creeps when I dont use the vga side functions but with them it is breezing along

https://cs.csub.edu/~msarr/pico6502/picocomputer_screensaver.mp4

**Comment by rumbledethumps** [comment]
What is was really like: https://www.youtube.com/watch?v=6x3be3xMx4w
I wonder if there's a de facto 2D API for the 6502. Maybe GEOS?

**Reply by michaelsarr** [reply]
yep.. that is what it was like back in the day, I didnt have any fancy Epyx basic on the C64 either

you could get sprites zipping around the screen but other stuff took a lot of voodoo

Atari basic had some primitive graphics functions built in

when I do the pixel calculations on the 6502 it runs more like that 

https://cs.csub.edu/~msarr/pico6502/screensaver_all_on_6502.mp4

**Comment by michaelsarr** [comment]
i may get the clock setting bit rolled into what i have so i can get something to quantify and try to optimize a couple functions i have for speed on the 6502. it will help me get familiar with doing assembly. I haven't written any in 40 years.... that's not a fun realization
